### PR TITLE
DW-271: Added methods for creating parent and sub case

### PR DESCRIPTION
**Pull request by jekuaitk**
https://jira.itkdev.dk/browse/DW-271

* Adds methods for creating cases and 'sub cases' in GetOrganized.

### CODE REVIEW

File: src/Service/Cases.php [hunk 1]
```diff
@@ -53,7 +53,7 @@ public function FindCases(array $query)
     }
 
     /**
-     * Example query:.
+     * Example query.
```
Comment (rimi-itk) [inline src/Service/Cases.php]: We should keep the colon rather than the period.

File: src/Service/Cases.php [hunk 2]
```diff
@@ -96,4 +96,52 @@ public function getByCaseId(string $caseId, string $caseTypePrefix = null): ?arr
 
         return $cases['CasesInfo'][0] ?? null;
     }
+
+    /**
+     * Example query.
```
Comment (rimi-itk) [inline src/Service/Cases.php]: Use colon here.

File: src/Service/Cases.php [hunk 3]
```diff
@@ -96,4 +96,52 @@ public function getByCaseId(string $caseId, string $caseTypePrefix = null): ?arr
...
+    {
+        return $this->getData(
+            'POST',
+            $this->getApiBasePath(),
+            ['json' => $query],
+        );
+    }
+
+    /**
+     * Example query.
```
Comment (rimi-itk) [inline src/Service/Cases.php]: Use colon here.

File: src/Service/Cases.php [hunk 4]
```diff
@@ -96,4 +96,52 @@ public function getByCaseId(string $caseId, string $caseTypePrefix = null): ?arr
...
+     *     "<z:row xmlns:z=\"#RowsetSchema\"
+     *      ows_Title=\"0123456789 - Test borger\"
+     *      ows_CCMContactData=\"Test borger;#;#0123456789;#;#\"
+     *      ows_CCMContactData_CPR=\"0123456789\"
+     *      ows_CaseStatus=\"Åben\"
+     *      />",
+     *  'ReturnWhenCaseFullyCreated' => true,
+     * ];
+     */
+    public function CreateCase(array $query): ?array
```
Comment (rimi-itk) [inline src/Service/Cases.php]: The parameter should not be called `$query`, but rather `$data` or something similar. “Query” is used for querying and not for creating stuff.

File: src/Service/Cases.php [hunk 5]
```diff
@@ -96,4 +96,52 @@ public function getByCaseId(string $caseId, string $caseTypePrefix = null): ?arr
...
+     *     "<z:row xmlns:z=\"#RowsetSchema\"
+     *     ows_Title=\"Undersag - test\"
+     *     ows_CCMParentCase=\"BOR-2022-000038\"
+     *     ows_ContentTypeId=\"0x0100512AABDB08FA4fadB4A10948B5A56C7C01\"
+     *     ows_CCMContactData_CPR=\"0123456789\"
+     *     ows_CaseStatus=\"Åben\"/>",
+     *  'ReturnWhenCaseFullyCreated' => true,
+     *  ];
+     */
+    public function CreateSubCase(array $query): ?array
```
Comment (rimi-itk) [inline src/Service/Cases.php]: The parameter should not be called `$query`, but rather `$data` or something similar. “Query” is used for querying and not for creating stuff.

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Some minor punctuation and language changes requested.

File: src/Service/Cases.php (hunk 1 above)
Comment (jekuaitk) [reply src/Service/Cases.php]: Updated method documentation.

File: src/Service/Cases.php (hunk 2 above)
Comment (jekuaitk) [reply src/Service/Cases.php]: Updated method documentation.

File: src/Service/Cases.php (hunk 3 above)
Comment (jekuaitk) [reply src/Service/Cases.php]: Updated method documentation.

File: src/Service/Cases.php (hunk 4 above)
Comment (jekuaitk) [reply src/Service/Cases.php]: Updated variable name.

File: src/Service/Cases.php (hunk 5 above)
Comment (jekuaitk) [reply src/Service/Cases.php]: Updated variable name.

File: src/Service/Cases.php [hunk 6]
```diff
@@ -28,12 +28,12 @@ protected function getApiBasePath(): string
      *
      * @throws GetOrganizedClientException
      */
-    public function FindCases(array $query)
+    public function FindCases(array $data)
```
Comment (rimi-itk) [inline src/Service/Cases.php]: The parameter should not be called `$data`, but rather `$query`. “Data” is used for creating stuff and not for querying.

File: src/Service/Cases.php [hunk 7]
```diff
@@ -53,24 +53,25 @@ public function FindCases(array $query)
...
      *       'InternalName' => 'ows_CaseID',
      *       'Value' => 'GEO-2022-033830'
      *     ],
      *   ],
      *   'CaseTypePrefixes' => ['GEO'],
-     * ]
+     * ];
      */
-    public function FindByCaseProperties(array $query): array
+    public function FindByCaseProperties(array $data): array
```
Comment (rimi-itk) [inline src/Service/Cases.php]: The parameter should not be called `$data`, but rather `$query`. “Data” is used for creating stuff and not for querying.

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Some linguistic changes requested.

File: src/Service/Cases.php (hunk 6 above)
Comment (jekuaitk) [reply src/Service/Cases.php]: Updated.

File: src/Service/Cases.php (hunk 7 above)
Comment (jekuaitk) [reply src/Service/Cases.php]: Updated.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**